tests/acquia_lift.test

```diff
@@ -3219,32 +3219,168 @@ class AcquiaLiftWebTestCampaignWizard extends AcquiaLiftWebTestBase {
     $this->assertNoText('Element form here');
 
     $edit = array(
-      'option_set_type' => 'block',
+      'variations[add_variation][option_set_types][option_set_type]' => 'block',
     );
 
-    $this->drupalPostAJAX(NULL, $edit, 'option_set_type');
-    $this->assertText('Block form here');
+    $this->drupalPostAJAX(NULL, $edit, 'variations[add_variation][option_set_types][option_set_type]');
+    $this->assertFieldByName('variations[add_variation][details][block][content][title]');
     $this->assertNoText('Element form here');
 
     $this->drupalGet('admin/structure/personalize/manage/' . $agent->getMachineName() . '/variations');
     // Actual forms should not be shown yet.
-    $this->assertNoText('Block form here');
+    $this->assertNoFieldByName('variations[add_variation][details][block][content][title]');
     $this->assertNoText('Element form here');
 
     $edit = array(
-      'option_set_type' => 'element',
+      'variations[add_variation][option_set_types][option_set_type]' => 'element',
     );
 
-    $this->drupalPostAJAX(NULL, $edit, 'option_set_type');
-    $this->assertNoText('Block form here');
+    $this->drupalPostAJAX(NULL, $edit, 'variations[add_variation][option_set_types][option_set_type]');
+    $this->assertNoFieldByName('variations[add_variation][details][block][content][title]');
     $this->assertText('Element form here');
   }
+
+  /**
+   * Test variation set creation.
+   */
+  function testWhatForm() {
+    $this->drupalLogin($this->managerUser);
+    $agent = $this->createTestAgent();
+    // Create a personalized block from existing blocks.
+    $this->drupalGet('admin/structure/personalize/manage/' . $agent->getMachineName() . '/variations');
+    // Indicate a block variation set type.
+    $edit = array(
+      'variations[add_variation][option_set_types][option_set_type]' => 'block',
+    );
+    $this->drupalPostAJAX(NULL, $edit, 'variations[add_variation][option_set_types][option_set_type]');
+
+    $edit = array(
+      'variations[add_variation][option_set_types][option_set_type]' => 'block',
+      'variations[add_variation][details][block][content][title]' => 'Muppets',
+      'variations[add_variation][details][block][content][pblock_wrapper][blocks][0][block][bid]' => 'comment_delta_recent',
+      'variations[add_variation][details][block][content][pblock_wrapper][blocks][1][block][bid]' => 'node_delta_recent',
+    );
+    $this->drupalPost(NULL, $edit, $this->getButton('wizard_next'));
+
+    $option_sets = personalize_option_set_load_by_agent($agent->getMachineName());
+    $this->assertEqual(count($option_sets), 1);
+
+    $option_set = reset($option_sets);
+    $this->assertEqual(count($option_set->options), 2);
+    $expected_queue_items[] = array(
+      'method' => 'savePoint',
+      'args' => array(
+        $agent->getMachineName(),
+        personalize_get_decision_point_name_for_option_set($option_set),
+      ),
+      'agent' => $agent->getMachineName(),
+    );
```

Comment:
Hmm, we should have a restriction in place ensuring this new workflow can only be used with the acquia_lift_target agent type (and if you create a new agent it can only be of this type) and then there won't be any items added to the queue here. Not sure if that's an easy restriction to put in at this point?